src/borrow.rs

```diff
@@ -69,6 +69,10 @@ impl<T> AtomicRefCell<T> {
         loop {
             let read = self.borrow_state.load(std::sync::atomic::Ordering::SeqCst);
             if read < 0 {
+                println!(
+                    "resource already borrowed as mutable: {}",
+                    std::any::type_name::<T>()
+                );
```

Comment:
The library should not be writing straight into stdout, rather eventing like this should be traced. In this case (and that below), I am not sure we should be both printing an error and duplicating that message with an `Err`. The callee is going to have to decide how to deal with the `Err` one way or another already.

Reply:
I do this print here because the error is a static lifetime str. That means we cannot return the actually type_name via stable right; const_type_name is an unstable nightly feature at the moment.

So would you purpose we trace the type name, and the Err is genericly named, or what other solution perhaps? I consider this an error case, hence the println

Reply:
We can change the error to an owned string (or wrap it in a proper error type enum). Most callees are going to immediately unwrap the result, which will panic with the message anyway.